tests/functional/test_list.py

```diff
@@ -543,3 +544,21 @@ def test_list_path_multiple(tmpdir, script, data):
     json_result = json.loads(result.stdout)
     assert {'name': 'simple', 'version': '2.0'} in json_result
     assert {'name': 'simple2', 'version': '3.0'} in json_result
+
+
+def test_list_skip_work_dir_pkg(script):
+    """
+    Test that list should not include package in working directory
+    """
+
+    # Create a test package and create .egg-info dir
+    pkg_path = create_test_package_with_setup(script,
+                                              name='simple',
+                                              version='1.0')
+    script.run('python', 'setup.py', 'egg_info',
+               expect_stderr=True, cwd=pkg_path)
+
+    # List should not include package simple when run from package directory
+    result = script.pip('list', '--format=json', cwd=pkg_path)
+    json_result = json.loads(result.stdout)
+    assert {'name': 'simple', 'version': '1.0'} not in json_result
```

Comment:
@deveshks you can do a variant of this test, that runs pip list with `PYTHONPATH=.` in environment. In that case `simple` must be in list result.

Reply:
Thanks @sbidoul , 

Could you help me out in how can I run pip list with `PYTHONPATH=.` in the test env? I tried `os.environ.update({'PYTHONPATH':'.'}` before running the list command but that didn't work.

I am also unable to find an example to run a pip command like `PYTHONPATH=. pip list` using the options in the test environment

Reply:
@deveshks `script.environ` should be what you are looking for.

Reply:
Thanks @sbidoul , That worked indeed. I have now added the test with the logic you mentioned. Please take a look.